### PR TITLE
add missing psp e2e tests for runasgroup

### PR DESCRIPTION
**Pull request by krmayankk**
```release-note
NONE
```

### CODE REVIEW

File: test/e2e/auth/pod_security_policy.go [hunk 1]
```diff
@@ -186,6 +186,21 @@ func testPrivilegedPods(tester func(pod *v1.Pod)) {
 		sysadmin.Spec.Containers[0].SecurityContext.AllowPrivilegeEscalation = nil
 		tester(sysadmin)
 	})
+
+	By("Running a RunAsGroup pod", func() {
+		sysadmin := restrictedPod("runasgroup")
+		gid := int64(65529)
```
Comment (tallclair) [inline test/e2e/auth/pod_security_policy.go]: nit: use GID 0 so it's more clear what this is doing

File: test/e2e/auth/pod_security_policy.go [hunk 2]
```diff
@@ -245,6 +260,7 @@ func restrictedPod(name string) *v1.Pod {
 				SecurityContext: &v1.SecurityContext{
 					AllowPrivilegeEscalation: boolPtr(false),
 					RunAsUser:                utilpointer.Int64Ptr(65534),
+					RunAsGroup:               utilpointer.Int64Ptr(65534),
```
Comment (tallclair) [inline test/e2e/auth/pod_security_policy.go]: nit: Can you make 65534 a constant, and replace all uses in this file? ("nobodyUser")

**Comment by jingweno** [comment]
:+1: Just bumped into this with group always being root...

**Pull request merged**